src/app/lib/business/logger/logger_manager.dart

```diff
@@ -0,0 +1,217 @@
+import 'dart:io';
+
+import 'package:app/access/logger/logger_repository.dart';
+import 'package:app/business/logger/custom_console_output.dart';
+import 'package:app/business/logger/custom_file_output.dart';
+import 'package:app/business/logger/level_log_filter.dart';
+import 'package:flutter_dotenv/flutter_dotenv.dart';
+import 'package:logger/logger.dart';
+import 'package:path_provider/path_provider.dart';
+import 'package:share_plus/share_plus.dart';
+
+/// Service that managed everything related to logging.
+abstract interface class LoggerManager {
+  factory LoggerManager(LoggerRepository loggerRepository) = _LoggerManager;
+
+  /// Gets whether console logging is enabled.
+  bool get isConsoleLoggingEnabled;
+
+  /// Gets whether file logging is enabled.
+  bool get isFileLoggingEnabled;
+
+  /// Gets whether logging configuration been changed via either [setIsConsoleLoggingEnabled] or [setIsFileLoggingEnabled].
+  bool get hasConfigurationBeenChanged;
+
+  /// Create an instancee of [Logger].
+  Future<Logger> createLogInstance();
+
+  // Deletes log file
+  Future<bool> deleteLogFile();
+
+  // Shares log file via email.
+  Future<bool> shareLogFile();
```

Comment:
These should probably have /// instead of // like the other comments above the other methods